paasta_tools/mesos_tools.py

```diff
@@ -655,21 +667,34 @@ def terminate_framework(framework_id):
     resp.raise_for_status()
 
 
-def get_container_name(app_id, slave_hostname=None, task_id=None):
+def get_tasks_from_app_id(app_id, slave_hostname=None):
     tasks = get_running_tasks_from_active_frameworks(app_id)
-    try:
-        if task_id:
-            task = next(task for task in tasks if task['id'] == task_id)
-        elif slave_hostname:
-            task = next(task for task in tasks if slave_hostname in task.slave['hostname'])
-        else:
-            raise TaskNotFound("Couldn't find task")
-    except StopIteration:
-        raise TaskNotFound("Couldn't find task")
-    container_name = "mesos-{0}.{1}".format(task.executor['tasks'][0]['slave_id'],
-                                            task.executor['container'])
-    return container_name
+    if slave_hostname:
+        tasks = filter_tasks(tasks, filter_task_by_hostname, hostname=slave_hostname)
+    return tasks
+
+
+def get_task(task_id, app_id=''):
+    tasks = get_running_tasks_from_active_frameworks(app_id)
```

Comment:
I think I want to leave this because it has a desired side effect: if you pass the wrong task_id for the service and instance then this will return an empty list and you get a useful error message.